tfjs-core/src/types.ts

```diff
@@ -196,3 +196,26 @@ export interface WebGPUData {
   buffer: GPUBuffer;
   zeroCopy?: boolean;
 }
+
+export enum GPUDataType {
+  Unknown = 0,
+  WebGL = 1,
+  WebGPU = 2,
+}
+
+export function getGPUDataType(values: TensorLike|WebGLData|
+                               WebGPUData): GPUDataType {
+  if (typeof values !== 'object') {
+    // Unknown data type.
+    return GPUDataType.Unknown;
+  }
+  if ('texture' in values) {
+    return GPUDataType.WebGL;
+  } else if (
+      'buffer' in values && !(values.buffer instanceof ArrayBuffer) &&
+      !(values.buffer instanceof SharedArrayBuffer)) {
+    return GPUDataType.WebGPU;
```

Comment:
Great catch, thanks you!

I am considering, rather than excluding cases, what about we do checks like:?
```js
if ('texture' in values && values.texture instanceof WebGLTexture) {
  return GPUDataType.WebGL;
} else if ( 'buffer' in values && values.buffer instanceof GPUBuffer) {
...
}
```

Reply:
@Linchenn, done. PTAL.

BTW, I am trying to add a test case 'tensor from ShardArrayBuffer', however, chrome complains 'ReferenceError: SharedArrayBuffer is not defined.' It seems 'yarn karma start' doesn't support SharedArrayBuffer .  This can be workaround by https://github.com/Honry/webnn-samples/blob/tflite-deeplab/server.js. 

@qjia7 @xhcao @haoyunfeix @gyagp , PTAL

Reply:
What about [this idiom](https://github.com/tensorflow/tfjs/blob/50f2a824dd901a4b8a4db883fb8f88d9525adde6/tfjs-core/src/ops/browser.ts#L86) (though I don't feel parenthesis is needed around the operand of `typeof`)?

Reply:
Thanks, @hujiajie, it seems the bots turrn green.

@Linchenn , Could you please take another look?